### ISSUE TITLE
Classic async reconstruction drops user stores in loop and sequential recipes

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#4817]
## Problem

The existing classic-async `TryBuildLoop` and `TryBuildSequentialVoid` recipes accept `MoveNext` bodies containing user field stores that the reconstructed body does not emit. The methods are reported as reconstructed at Full fidelity while the side effect disappears.

Reproduced on #4472 local head `cf5f91029` with compiler-produced `runtime-async=off` fixtures:

```csharp
public static async Task<int> LoopWithFieldStore(Task<int>[] tasks)
{
    int sum = 0;
    foreach (var task in tasks)
    {
        sum += await task;
        Observed = sum;
    }
    return sum;
}
```

Current output:

```csharp
int sum = 0;
foreach (Task<int> task in tasks)
{
    sum += (await task);
}
return sum;
```

And:

```csharp
public static async Task SequentialWithFieldStore(Task<int> a, Task<int> b)
{
    int alpha = await a;
    Observed = alpha;
    int beta = await b;
    GC.KeepAlive((alpha, beta));
}
```

Current output:

```csharp
int alpha = await a;
int y = await b;
GC.KeepAlive((alpha, y));
```

The field store is absent in both outputs. The sequential recipe also hard-codes the second local name as `y` instead of consuming the available symbol name (`beta`) or leaving it unnamed.

## Required behavior

Until the region/realization ledger proves and emits these effects, reconstruction must decline honestly with `UnconsumedExecutionRegion`; it must not emit a plausible body that drops them. Existing accepted fixtures without extra user effects must remain accepted.

Local names must come from imported symbol facts when available and remain absent otherwise; the recipe must not invent a source name.

This is directly gated by #4472 slice 0 region ownership.

**Comment by richlander** [comment]
Reproduced independently in the #4472 worktree and fixed locally in commit `06588b560`.

The slice-0 planner now rejects execution snapshots containing stores outside the exact state-machine type before any recipe can accept them. Both compiler-produced witnesses return `Declined(UnconsumedExecutionRegion, ReplacedNarrowHandoff)` with a visible unsupported marker; the existing loop and sequential fixtures remain reconstructed. The sequential recipe now consumes `moveNext.LocalNames` for the second local and leaves it unnamed when symbols are absent instead of inventing `y`.

This remains local/unpushed pending the complete #4472 region/realization gate.
